platform/ui/src/views/StudyList/components/StudyListFilter.js

```diff
@@ -1,138 +1,28 @@
-        sortBy: _sortDirection !== 0 ? name : '',
-        sortDirection: _sortDirection,
-      }));
-    }
-  };
-
-  const clearFilters = () => {
-    setcurrentFiltersValues(filtersValues);
-  };
-
-  const renderFieldInputComponent = ({
-    name,
-    displayName,
-    inputProps,
-    isSortable,
-    inputType,
-  }) => {
-    const _isSortable = isSortable && numOfStudies <= 100 && numOfStudies > 0;
-    const _sortDirection = sortBy !== name ? 'none' : sortDirection;
-    const onLabelClick = () => {
-      if (_isSortable) {
-        handleFilterLabelClick(name);
-      }
-    };
-    switch (inputType) {
-      case 'Text':
-        return (
-          <InputText
-            key={name}
-            label={displayName}
-            isSortable={_isSortable}
-            sortDirection={_sortDirection}
-            onLabelClick={onLabelClick}
-            value={currentFiltersValues[name]}
-            onChange={newValue => {
-              setcurrentFiltersValues(prevState => ({
-                ...prevState,
-                [name]: newValue,
-              }));
-            }}
-          />
-        );
-        break;
-      case 'MultiSelect':
-        return (
-          <InputMultiSelect
-            key={name}
-            label={displayName}
-            isSortable={_isSortable}
-            sortDirection={_sortDirection}
-            onLabelClick={onLabelClick}
-            options={inputProps.options}
-            value={currentFiltersValues[name]}
-            onChange={newValue => {
-              setcurrentFiltersValues(prevState => ({
-                ...prevState,
-                [name]: newValue,
-              }));
-            }}
-          />
-        );
-      case 'DateRange':
-        return (
-          <InputDateRange
-            key={name}
-            label={displayName}
-            isSortable={_isSortable}
-            sortDirection={_sortDirection}
-            onLabelClick={onLabelClick}
-            value={currentFiltersValues[name]}
-            onChange={newValue => {
-              setcurrentFiltersValues(prevState => ({
-                ...prevState,
-                [name]: newValue,
-              }));
-            }}
-          />
-        );
-      case 'None':
-        return (
-          <InputLabelWrapper
-            key={name}
-            label={displayName}
-            isSortable={_isSortable}
-            sortDirection={_sortDirection}
-            onLabelClick={onLabelClick}
-          />
-        );
-      default:
-        break;
-    }
+const StudyListFilter = ({
+  filtersMeta,
+  filterValues,
+  setFilterValues,
+  clearFilters,
+  isFiltering,
+  numOfStudies,
+}) => {
+  const { sortBy, sortDirection } = filterValues;
+  const filterSorting = { sortBy, sortDirection };
+  const setFilterSorting = sortingValues => {
+    setFilterValues({
+      ...filterValues,
+      ...sortingValues,
+    });
   };
+  const isSortingEnable = numOfStudies > 0 && numOfStudies <= 100;
```

Comment:
Maybe it's a good time to define a prop that enables/disable the sorting, and let the application to deal with this, instead of hard-coding this logic here. What do you think?

Reply:
Ideally this StudyListFilter will do not exist in our future implementation and this rule will become part of the studyList page.